Realm/RLMSyncSession.mm

```diff
@@ -24,10 +24,42 @@
 
 using namespace realm;
 
-@interface RLMSyncSession () {
+@interface RLMProgressNotificationToken() {
+    uint64_t _token;
     std::weak_ptr<SyncSession> _session;
 }
+@end
+
+@implementation RLMProgressNotificationToken
+
+- (void)stop {
+    if (auto session = _session.lock()) {
+        session->unregister_progress_notifier(_token);
+    }
+}
 
+- (void)dealloc {
```

Comment:
So the expected behavior is, upon dealloc, if `-stop` hasn't been called previously a `NSLog` warning message is fired?